### PR TITLE
Make savemetric work with em protocol

### PR DESCRIPTION
**Pull request by xiki-tempula**
FreeEnergy/_relative._analyse_noSOMD will now prioritise PARQUET files.
Amber and gromacs process saveMetric method can now handle columns with not the same number of values.

### CODE REVIEW

File: python/BioSimSpace/Sandpit/Exscientia/FreeEnergy/_relative.py [hunk 1]
```diff
@@ -626,29 +626,48 @@ def _analyse_noSOMD(
...
+                software="PARQUET",
+                dir=work_dir,
+                prefix="/lambda_*/" + prefix,
+                suffix="parquet",
+                T=temperature / _Units.Temperature.kelvin,
+                outdirectory=work_dir,
+                **kwargs,
+            )
+            workflow.run(estimators=estimator, breakdown=None, forwrev=None, **kwargs)
+        except:
```
Comment (msuruzhon) [inline python/BioSimSpace/Sandpit/Exscientia/FreeEnergy/_relative.py]: Less general except?

File: python/BioSimSpace/Sandpit/Exscientia/Process/_amber.py [hunk 2]
```diff
@@ -26,6 +26,7 @@
 
 __all__ = ["Amber"]
 
+import warnings
```
Comment (msuruzhon) [inline python/BioSimSpace/Sandpit/Exscientia/Process/_amber.py]: I guess this should be `import warnings as _warnings`?

File: python/BioSimSpace/Sandpit/Exscientia/Process/_amber.py [hunk 3]
```diff
@@ -2494,27 +2495,45 @@ def saveMetric(
...
+            length_dict = {key: len(value) for key, value in datadict.items()}
+            warnings.warn(f'Not all metric has the same number of data points ({length_dict}).'
+                          f'All columns will be truncated the same length.')
+            length = min(length_dict.values())
+            new_datadict = {key: value[:length] for key, value in datadict.items()}
+            df = pd.DataFrame(data=new_datadict)
         df = df.set_index("Time (ps)")
         df.to_parquet(path=f"{self.workDir()}/{filename}", index=True)
-        if isinstance(self._protocol, _Protocol._FreeEnergyMixin):
+        if isinstance(self._protocol, _Protocol.FreeEnergy):
```
Comment (msuruzhon) [inline python/BioSimSpace/Sandpit/Exscientia/Process/_amber.py:2535]: Why is that changed?

Comment (xiki-tempula) [reply python/BioSimSpace/Sandpit/Exscientia/Process/_amber.py:2535]: Because neither FreeEnergyEquilibration nor FreeEnergyMinimisation generate sensible energy files

**Review by msuruzhon — APPROVED** [review]
Couple of small changes but LGTM otherwise
